src/annotator/guest.ts

```diff
@@ -330,15 +336,27 @@ export class Guest extends TinyEmitter implements Annotator, Destroyable {
     // them from propagating out of their shadow roots, and hence clicking on
     // elements in the sidebar's vertical toolbar or adder won't close the
     // sidebar.
-    const maybeCloseSidebar = (element: Element) => {
+    const maybeCloseSidebar = (event: PointerEvent) => {
       // Don't hide the sidebar if event was disabled because the sidebar
       // doesn't overlap the content.
       if (this._integration.sideBySideActive()) {
         return;
       }
 
       // Don't hide the sidebar if the event comes from an element that contains a highlight
-      if (annotationsAt(element).length) {
+      if (annotationsAt(event.target as Element).length) {
+        return;
+      }
+
+      // If the click is within the bounds of the sidebar, ignore it. We don't
+      // want to close the sidebar if the user clicks eg. in transparent areas
+      // of the toolbar / bucket bar along the edge. Clicks within the sidebar
+      // iframe won't be received by the guest frame(s) at all.
+      if (
+        frameFillsAncestor(window, this._hostFrame) &&
+        this._sidebarLayout?.expanded &&
+        window.innerWidth - event.clientX < this._sidebarLayout.width
+      ) {
         return;
       }
```

Comment:
The `frameFillsAncestor` check is used in a couple of places in `Guest` to only apply certain logic if the guest frame is either the same as the host frame, or fills most of the area of the host frame (as it does in eg. VitalSource ebooks).